azurerm/internal/services/managedapplication/tests/resource_arm_managed_application_definition_test.go

```diff
@@ -0,0 +1,326 @@
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.ENV", "Test"),
+				),
+			},
+			data.ImportStep("create_ui_definition", "main_template", "package_file_uri"),
+		},
+	})
+}
+
+func testCheckAzureRMManagedApplicationDefinitionExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Managed Application Definition not found: %s", resourceName)
+		}
+
+		id, err := parse.ManagedApplicationDefinitionID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).ManagedApplication.ApplicationDefinitionClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Managed Application Definition %q (Resource Group %q) does not exist", id.Name, id.ResourceGroup)
+			}
+			return fmt.Errorf("Bad: Get on ManagedApplication.ApplicationDefinitionClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMManagedApplicationDefinitionDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).ManagedApplication.ApplicationDefinitionClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_managed_application_definition" {
+			continue
+		}
+
+		id, err := parse.ManagedApplicationDefinitionID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on ManagedApplication.ApplicationDefinitionClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMManagedApplicationDefinition_basic(data acceptance.TestData) string {
+	template := testAccAzureRMManagedApplicationDefinition_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_managed_application_definition" "test" {
+  name                = "acctestAppDef%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  lock_level          = "ReadOnly"
+  package_file_uri    = "https://github.com/Azure/azure-managedapp-samples/raw/master/Managed Application Sample Packages/201-managed-storage-account/managedstorage.zip"
+  display_name        = "TestManagedApplicationDefinition"
+  description         = "Test Managed Application Definition"
+  package_enabled     = false
+
+  authorization {
+    service_principal_id = data.azurerm_client_config.current.object_id
+    role_definition_id   = "b24988ac-6180-42a0-ab88-20f7382dd24c"
```

Comment:
Use data source `azurerm_role_definition` instead, to retrieve the id.

Reply:
updated